src/runtime/openglcompute.cpp

```diff
@@ -339,6 +340,29 @@ WEAK int halide_openglcompute_device_free(void *user_context, halide_buffer_t *b
     return 0;
 }
 
+namespace {
+
+template <typename Source, typename Dest>
+__attribute__((always_inline)) void converting_copy_memory_helper(const device_copy &copy, int d, int64_t src_off, int64_t dst_off) {
+    // Skip size-1 dimensions
+    while (d >= 0 && copy.extent[d] == 1) d--;
+
+    if (d == -1) {
+        const Source *from = (Source *)(copy.src + src_off);
+        Dest *to = (Dest *)(copy.dst + dst_off);
+        for (uint64_t index = 0; index < copy.chunk_size; index++) {
```

Comment:
Would it be profitable to just use memcpy() here instead?

Reply:
There's actually a type conversion happening so memcpy won't work.